### DISCUSSION TITLE
How use locale_str in hybrid_group

### DISCUSSION THREAD

**Discussion by riquedev** [Rapptz/discord.py#9521] [Q&A] [answered]
Hello, I managed to translate my commands using app_commands.locale_str and making a class app_commands.Translator itself, however in the case of hybrid_group the translation of their name is not working correctly, for example:

`/gather wood`

After translating it should be (pt-br):

`/coletar madeira`

But I end up with the result:

`/gather madeira`

Any tips on how to resolve this?

```python
@commands.hybrid_group(name=app_commands.locale_str('gather'))
async def gather(self, ctx: discord.Interaction):
    pass

@gather.command(name=app_commands.locale_str('wood'))
@app_commands.checks.cooldown(6, 300)
async def wood(self, ctx: commands.Context):
     pass
```

**Comment by Rapptz** [comment]
Your name shouldn't have spaces in it, though I can't immediately see anything that would cause the `name` to not be passed to the translator.

**Comment by riquedev** [comment]
> Your name shouldn't have spaces in it, though I can't immediately see anything that would cause the `name` to not be passed to the translator.

I corrected, there is no that space it's just the "wood"

**Comment by riquedev** [comment, marked as answer]
I found the problem, for some reason when the hybrid_group passes to the Translator it doesn't come with a context of "TranslationContextLocation.command_name".
In my code I observe these contexts to use in GetText (because my translations are in YML files), so it always ended up falling outside the conditionals, to correct I just added an else testing all possibilities, it's not the best way to do it but it worked...

```python
from __future__ import annotations

import discord
from discord import app_commands
from discord.app_commands import TranslationContextLocation

from IsekaiTensei.discord_bot.utils.get_text import GetText


class IsekaiTranslator(app_commands.Translator):
    async def translate(
            self,
            string: app_commands.locale_str,
            locale: discord.Locale,
            context: app_commands.TranslationContext,
    ) -> str | None:
        try:
            if context.location in (TranslationContextLocation.command_name,
                                    TranslationContextLocation.command_description):
                return GetText(context, locale).message(str(string))
            elif context.location in (TranslationContextLocation.parameter_name,
                                      TranslationContextLocation.parameter_description):
                return GetText(context, locale).parameter(str(string))
            elif context.location in (TranslationContextLocation.choice_name,):
                return GetText(context, locale).choice(str(string))
            else:
                value = None
                for method in [GetText(context, locale).message,
                               GetText(context, locale).parameter,
                               GetText(context, locale).choice]:
                    value = method(str(string))
                    if value is not None:
                        break
                return value
        except Exception as er:
            print(er)
            return str(string)

```

**Reply by Rapptz** [reply]
Well of course not, the context for groups would be `group_name` not `command_name`.

**Reply by riquedev** [reply]
You're right, I ended up not noticing this I'm going to remove this loop from the else
